projects/ad400x/common/ad400x_bd.tcl

```diff
@@ -31,14 +31,15 @@ current_bd_instance /spi_ad400x
   ## to setup the sample rate of the system change the PULSE_PERIOD value
   ## the acutal sample rate will be PULSE_PERIOD * (1/sys_cpu_clk)
   ## fsys_cpu_clk must be 100 MHz
-  ## Default tCYC = 560 ns
-  ad_ip_parameter trigger_gen CONFIG.PULSE_PERIOD 56
+  set cycle_per_sec_100mhz 100000000
+  set sampling_cycle [expr $cycle_per_sec_100mhz / $adc_sampling_rate]
```

Comment:
This is rounding down the duration, which means it is rounding up the sampling rate. If we want to make sure that the sampling rate is always <= than the specified one we could use

int(ceil(double($cycle_per_sec_100mhz) / $adc_sampling_rate))

This for example yields 56 for adc_sampling_rate = 1800000, instead of 55 when rounding down.

Reply:
Yes, you're right. Thanks!